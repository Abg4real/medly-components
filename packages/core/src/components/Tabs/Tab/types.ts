import { HTMLProps } from '@medly-components/utils';
import { TabSize } from '../types';

export interface StyledProps extends HTMLProps<HTMLButtonElement> {
    active?: boolean;
    hasIcon?: boolean;
    tabSize?: TabSize;
}

export interface Props extends StyledProps {
    /** Id of tab */
    id: any;
    /** Label of the tab */
    label: string;
    /** To be used to hide the tab */
    hide?: boolean;
    /** Icon to be shown */
    icon?: React.ElementType;
}
