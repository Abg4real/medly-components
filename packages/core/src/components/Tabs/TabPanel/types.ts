import { HTMLProps } from '@medly-components/utils';

export interface Props extends HTMLProps<HTMLDivElement> {
    active: any;
    forceRender?: boolean;
}

export interface StyledProps extends HTMLProps<HTMLDivElement> {
    isActive: boolean;
}
