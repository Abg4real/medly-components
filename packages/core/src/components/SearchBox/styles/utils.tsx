import { SvgIcon } from '@medly-components/icons';
import { css } from '@medly-components/utils';
import { Props } from '../types';

export const getSearchBoxSize = ({ theme, size }: Props) => theme.searchBox.sizes[size].height;

export const getBorderAndBoxShadow = (color: string, boxShadow: string) => {
    return css`
        border-color: ${color};
        box-shadow: ${boxShadow};
    `;
};

export const getIconWrapperStyle = ({ theme, size }: Props) => {
    return css`
        width: ${getSearchBoxSize};
        min-width: ${getSearchBoxSize};
        height: auto;
        display: inline-flex;
        align-items: center;
        justify-content: center;
        ${SvgIcon} {
            cursor: pointer;
            font-size: ${theme.searchBox.sizes[size].iconSize};
        }
    `;
};
