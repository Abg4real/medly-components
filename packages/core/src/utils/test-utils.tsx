import { defaultTheme } from '@medly-components/theme';
import React from 'react';
import { render, RenderResult } from 'react-testing-library';
import { ThemeProvider } from './styled';

const AllTheProviders: React.FunctionComponent = props => (
    <ThemeProvider theme={defaultTheme}>
        <>{props.children}</>
    </ThemeProvider>
);

const customRender = (ui: React.ReactElement<any>, options?: object): RenderResult => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from 'react-testing-library';

// override render method
export { customRender as render };
