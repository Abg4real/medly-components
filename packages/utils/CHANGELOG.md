# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [1.0.0-beta.16](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.15...@medly-components/utils@1.0.0-beta.16) (2020-04-11)


### Features

* add typography and refactor components based on new typography ([462f2aa](https://github.com/medlypharmacy/medly-components/commit/462f2aa5b24c968b941887e6fcea5fd4e5c7b3f1))





# [1.0.0-beta.15](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.14...@medly-components/utils@1.0.0-beta.15) (2020-04-02)


### Bug Fixes

* **all:** fix lint issues ([ca12080](https://github.com/medlypharmacy/medly-components/commit/ca1208092c7a4d0ce0e22f20cb31c0400c4ddbff))





# [1.0.0-beta.14](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.13...@medly-components/utils@1.0.0-beta.14) (2020-03-30)


### Features

* **all:** add rollup to create bundles ([08b5972](https://github.com/medlypharmacy/medly-components/commit/08b5972f4a221037d14449ffce508f8656d0f447))





# [1.0.0-beta.13](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.12...@medly-components/utils@1.0.0-beta.13) (2020-03-29)

**Note:** Version bump only for package @medly-components/utils





# [1.0.0-beta.12](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.11...@medly-components/utils@1.0.0-beta.12) (2020-02-06)

**Note:** Version bump only for package @medly-components/utils





# [1.0.0-beta.11](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.10...@medly-components/utils@1.0.0-beta.11) (2020-02-04)

**Note:** Version bump only for package @medly-components/utils





# [1.0.0-beta.10](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.9...@medly-components/utils@1.0.0-beta.10) (2020-01-22)

**Note:** Version bump only for package @medly-components/utils





# [1.0.0-beta.9](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.8...@medly-components/utils@1.0.0-beta.9) (2020-01-21)


### Features

* **core:** add functionaity to accept date as string also in datepicker component ([3c3c377](https://github.com/medlypharmacy/medly-components/commit/3c3c377a66c470e6765d4a98f0f55b443b5ab0bc))
* **icons:** add bill icon ([2b461c2](https://github.com/medlypharmacy/medly-components/commit/2b461c2478590247498b08f13cad83d225efff6e))





# [1.0.0-beta.8](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.7...@medly-components/utils@1.0.0-beta.8) (2020-01-12)


### Features

* **all:** add side-effects to package.json ([0f78b4c](https://github.com/medlypharmacy/medly-components/commit/0f78b4c2ba432373b9d66861c25a265dacd5d433))





# [1.0.0-beta.7](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.6...@medly-components/utils@1.0.0-beta.7) (2020-01-09)

**Note:** Version bump only for package @medly-components/utils





# [1.0.0-beta.6](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.5...@medly-components/utils@1.0.0-beta.6) (2019-12-17)


### Bug Fixes

* **core:** fix modal close issue ([f2d8fe9](https://github.com/medlypharmacy/medly-components/commit/f2d8fe9553f9efdbeb96c8b5b213844f0c472777))





# [1.0.0-beta.5](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.4...@medly-components/utils@1.0.0-beta.5) (2019-12-10)

**Note:** Version bump only for package @medly-components/utils





# [1.0.0-beta.4](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.3...@medly-components/utils@1.0.0-beta.4) (2019-11-28)

**Note:** Version bump only for package @medly-components/utils





# [1.0.0-beta.3](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.2...@medly-components/utils@1.0.0-beta.3) (2019-11-13)

**Note:** Version bump only for package @medly-components/utils





# [1.0.0-beta.2](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.1...@medly-components/utils@1.0.0-beta.2) (2019-11-13)

**Note:** Version bump only for package @medly-components/utils





# [1.0.0-beta.1](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.2.0...@medly-components/utils@1.0.0-beta.1) (2019-11-13)


### Bug Fixes

* **all:** remove changelog for pre release setup ([0795f5c](https://github.com/medlypharmacy/medly-components/commit/0795f5c139b529c3f6b65442ff71a8826ecf2c56))
* fixed various issues in date picker ([00fed84](https://github.com/medlypharmacy/medly-components/commit/00fed842b6d363f817de2c8703785c62557cf50d))


### Features

* **all:** remove s3 upload command from all packages ([89ea085](https://github.com/medlypharmacy/medly-components/commit/89ea085082829cba2098342c9793a43e2bc2e24a))
* **core:** add capability to change option in Single select through arrow keys ([fd864b9](https://github.com/medlypharmacy/medly-components/commit/fd864b9eb103bc042086c51236f10222cfc57367))
* added capitalize function to capitalize first letter of the string ([2613604](https://github.com/medlypharmacy/medly-components/commit/261360434dcd02de4cde438350d00696323d16d5))
* added DatePicker and DateRangePicker component ([5dc6d38](https://github.com/medlypharmacy/medly-components/commit/5dc6d38f66f6d1d530f830da13a9624715f6f704))
