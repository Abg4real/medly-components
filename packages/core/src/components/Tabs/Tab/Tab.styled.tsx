import { SvgIcon } from '@medly-components/icons';
import { Theme } from '@medly-components/theme/src';
import { css, styled } from '@medly-components/utils';
import { TextStyled } from '../../Text/Text.styled';
import { TabSize } from '../types';
import { StyledProps } from './types';

const getTabSpacing = (tabSize: TabSize, hasIcon = false) => {
    const verticalSpacing = 15;
    let spacing = verticalSpacing;

    if (tabSize === 'L') {
        spacing = verticalSpacing + 8;
    }

    if (hasIcon) {
        return spacing - 1;
    }
    return spacing;
};

const activeStyle = (theme: Theme, active: boolean, tabSize: TabSize) => {
    const {
        tabs: { textColor, backgroundColor, backgroundTheme, tabStyle }
    } = theme;

    if (backgroundTheme === 'WHITE' && tabStyle === 'CLOSED') {
        return css``;
    } else {
        return css`
            color: ${textColor.active};
            border-bottom: 4px solid ${textColor.active};
            ${SvgIcon} {
                * {
                    fill: ${active && tabSize !== 'S' ? backgroundColor.nonActive : textColor.active};
                }
            }
            &:hover {
                background-color: ${backgroundColor.nonActive};
                ${TextStyled} {
                    color: ${textColor.active};
                }
            }
        `;
    }
};

const disabledStyle = (theme: Theme) => {
    return css`
        opacity: 0.4;
        &:hover {
            background-color: transparent;
            ${TextStyled} {
                color: ${theme.tabs.textColor.nonActive};
            }
        }
    `;
};

const nonActiveStyle = (theme: Theme) => {
    const {
        tabs: { textColor, backgroundColor }
    } = theme;
    return css`
        color: ${textColor.nonActive};
        &:hover {
            background-color: ${backgroundColor.hovered};
            ${TextStyled} {
                color: ${textColor.hovered};
            }
        }
    `;
};

const svgStyles = (theme: Theme, active: boolean) => {
    const {
        tabs: { textColor }
    } = theme;
    return css`
        ${SvgIcon} {
            border-radius: 50%;
            border: 1px solid ${active ? textColor.active : theme.colors.grey[100]};
            background-color: ${active ? textColor.active : theme.colors.grey[100]};
            padding: 5px;
            font-size: 2rem;
        }
    `;
};

export const Button = styled.button<StyledProps>`
    background-color: transparent;
    border: 0;
    border-bottom: 4px solid transparent;
    padding: ${({ tabSize, hasIcon }) => `${getTabSpacing(tabSize, hasIcon)}px ${16}px`};
    user-select: none;
    text-decoration: none;
    display: flex;
    align-items: center;
    flex-direction: row;
    justify-content: flex-start;

    ${TextStyled} {
        font-size: ${({ tabSize }) => (tabSize === 'S' ? '1.4rem' : '1.6rem')};
        margin-left: ${({ hasIcon }) => (hasIcon ? '16px' : 0)};
    }

    &:hover {
        cursor: pointer;
    }

    &:focus {
        outline: none;
    }

    &:disabled {
        cursor: not-allowed;
    }

    ${({ theme, active, tabSize, hasIcon }) => hasIcon && tabSize !== 'S' && svgStyles(theme, active)}

    ${({ theme, active }) => !active && nonActiveStyle(theme)}

    ${({ theme, active, tabSize }) => active && activeStyle(theme, active, tabSize)}

    ${({ theme, disabled }) => disabled && disabledStyle(theme)}
`;
