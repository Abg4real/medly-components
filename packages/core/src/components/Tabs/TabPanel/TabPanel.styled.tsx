import { styled } from '@medly-components/utils';

export const TabPanel = styled('div')`
    &:focus {
        outline: none;
    }
    padding: 10px;
`;
