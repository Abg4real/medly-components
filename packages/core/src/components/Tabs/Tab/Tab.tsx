import { WithStyle } from '@medly-components/utils';
import React from 'react';
import Text from '../../Text';
import * as Styled from './Tab.styled';
import { Props } from './types';

export const Tab: React.FC<Props> & WithStyle = React.memo(props => {
    const { id, active, label, icon: Icon = null, ...restProps } = props;

    return (
        <Styled.Button
            {...{ id, active, ...restProps }}
            type="button"
            role="tab"
            aria-selected={active ? 'true' : 'false'}
            aria-controls={`panel-${id}`}
            tabIndex={active ? 0 : -1}
        >
            {Icon && <Icon focusable={false} />}
            <Text>{label}</Text>
        </Styled.Button>
    );
});
Tab.displayName = 'Tab';
Tab.Style = Styled.Button;
