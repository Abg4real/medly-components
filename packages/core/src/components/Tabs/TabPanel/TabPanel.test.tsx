import { TestUtils } from '@medly-components/utils';
import React from 'react';
import Tab from '../Tab';
import { TabPanel } from './TabPanel';

const { render } = TestUtils;
describe('TabPanel', () => {
    it('should render only active tab content', () => {
        const { container } = render(
            <TabPanel id="tab-panel" active="tab1" onChange={jest.fn()}>
                <Tab id="tab1" label="Add">
                    Content for the add panel
                </Tab>
                <Tab id="tab2" label="Edit">
                    Content for the edit panel
                </Tab>
            </TabPanel>
        );
        expect(container).toMatchSnapshot();
    });
});
