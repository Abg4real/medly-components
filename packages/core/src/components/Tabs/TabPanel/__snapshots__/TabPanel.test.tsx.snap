// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabPanel should render only avtive tab content 1`] = `
.c0 {
  padding: 10px;
}

.c0:focus {
  outline: none;
}

<div>
  <div
    aria-labelledby="tab1"
    class="c0"
    id="panel-tab1"
    role="tabpanel"
    tabindex="0"
  >
    Content for the add panel
  </div>
</div>
`;
